mlflow/pyfunc/agent_server/__init__.py

```diff
@@ -0,0 +1,485 @@
+"""
+FastAPI-based server for hosting MLflow agents with multiple protocol support.
+
+This module provides a production-ready agent server that supports multiple agent types:
+- ResponsesAgent (agent/v1/responses): OpenAI-compatible responses format
+- ChatCompletion (agent/v1/chat): OpenAI chat completion format
+- ChatAgent (agent/v2/chat): MLflow's enhanced chat agent format
+
+Key Features:
+- Decorator-based function registration (@invoke, @stream) for easy agent development
+- Protocol-specific request/response validation using AgentValidator
+- Context-aware request header management for Databricks Apps authentication
+- Streaming and non-streaming response support with Server-Sent Events (SSE)
+- MLflow tracing integration with automatic span creation and attribute setting
+- Static file serving for optional agent UI components
+- Health check endpoint for monitoring
+
+Architecture:
+- AgentServer: Main FastAPI application with route setup and middleware
+- AgentValidator: Protocol-specific validation for requests and responses
+- Context isolation: Thread-safe request header management using contextvars
+- Function registration: Global decorators for invoke/stream endpoint functions
+
+Usage:
+    from mlflow.pyfunc.agent_server import AgentServer, invoke, stream
```

Comment:
`from mlflow.genai import AgentServer` or `from mlflow.genai import FastAgent` (similar to FastMCP/FastAPI import) would probably be better especially for folks unfamiliar with MLflow models, who won't know what pyfunc is